amethyst_rendy/src/pass/flat2d.rs

```diff
@@ -322,9 +337,21 @@ impl<B: Backend> RenderGroup<B, GraphAuxData> for DrawFlat2DTransparent<B> {
                     Some((
                         entity,
                         (
-                            world.get_component::<SpriteRender>(*entity)?,
-                            world.get_component::<LocalToWorld>(*entity)?,
-                            world.get_component::<Tint>(*entity),
+                            world
+                                .entry_ref(*entity)
+                                .unwrap()
+                                .into_component::<SpriteRender>()
+                                .ok()?,
+                            world
+                                .entry_ref(*entity)
+                                .unwrap()
+                                .into_component::<LocalToWorld>()
+                                .ok()?,
```

Comment:
I think we should make this a function, seeing how much duplication there is. 
Maybe `get_component::<Type>(world, entity)` ?

Reply:
This code would be really simplified with the query API `<(&SpriteRender, &LocalToWorld, Option<&Tint>)>::query().get(world, entity)`, but the performance is currently quite poor comapred to `entry_ref`. Ayfid mentioned that it will be improved in the future so I'm not sure whether to use it or not.

Query is currently about 2 times slower so maybe not a big deal, since it's linear time.